extensions/scarb-cairo-run/src/main.rs

```diff
@@ -0,0 +1,115 @@
+use std::collections::HashMap;
+use std::env;
+use std::fs;
+
+use anyhow::Result;
+use cairo_lang_runner::short_string::as_cairo_short_string;
+use cairo_lang_runner::{SierraCasmRunner, StarknetState};
+use cairo_lang_sierra::extensions::gas::{
+    BuiltinCostWithdrawGasLibfunc, RedepositGasLibfunc, WithdrawGasLibfunc,
+};
+use cairo_lang_sierra::extensions::NamedLibfunc;
+
+use camino::Utf8PathBuf;
+use clap::Parser;
+
+use scarb_metadata::packages_filter::PackagesFilter;
+use scarb_metadata::MetadataCommand;
+
+#[derive(Parser, Clone, Debug)]
+#[command(author, version)]
+struct Args {
+    #[command(flatten)]
+    packages_filter: PackagesFilter,
+    #[arg(long)]
+    available_gas: Option<usize>,
+    #[arg(long, default_value_t = false)]
+    print_full_memory: bool,
+}
+
+fn main() -> Result<()> {
+    let args: Args = Args::parse();
+    let metadata = MetadataCommand::new().inherit_stderr().exec()?;
+
+    let package = args.packages_filter.match_one(&metadata)?;
+    println!("running {} ...", package.name);
+
+    let filename = format!("{}.sierra", package.name);
+    let path = Utf8PathBuf::from(env::var("SCARB_TARGET_DIR").unwrap())
+        .join("dev")
+        .join(filename.clone());
+
+    if !path.exists() {
+        anyhow::bail!(format!(
+            "File {} does not exist. Run scarb build to compile all of this package's Cairo files.",
+            filename
+        ));
```

Comment:
```suggestion
        bail!(formatdoc! {r#"
            package has not been compiled, file does not exist: {filename}
            help: run `scarb build` to compile the package
        "#));
```

1. bail! does format! under the hood, but here I suggest using https://docs.rs/indoc/latest/indoc/macro.formatdoc.html
2. we follow Rust's error convention (unlike Cairo compiler internally), which means that errors start with lower case, are rather a single sentence and do not end with a dot
3. we also try to make our errors look like Rust's diagnostics, hence the `help` part